### PR TITLE
Move to NE 5.1.2 - Add Config for new POVs

### PR DESCRIPTION
**Pull request by travisgrigsby**
* #2074 - update to use NE 5.1.0
* #2075 - Add new NE TLC (top level countries) and ISO tables and remove support for admin-0-countries.
* #2081 - Add TLC POV for NE
* #2082 - Export `fclass_iso` and `fclass_tlc` and `label_x` and `label_y` for OSM countries that match NE countries

For #2082, I added the functionality to drop it into the output, but we aren't doing anything with it yet, so I haven't written any tests for it.  Once we decide to adopt a label_x and label_y if present, for example, there will be compelling function to test.

### CODE REVIEW

**Review by nvkelso — CHANGES_REQUESTED** [review]
Please update the versioned asset URLs to match the new pattern. #2074

And also add the new NE table to the data assets list and wire it into the NE tables. #2075

**Comment by travisgrigsby** [comment]
Updated for #2074

File: data/assets.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 bucket: nextzen-tile-assets
-datestamp: 20220426
+datestamp: 20220429
```
Comment (nvkelso) [inline data/assets.yaml:2]: NIt: but 5.1.0 was only officially released on May 4th, so bring this forward?

File: data/assets.yaml [hunk 2]
```diff
@@ -55,98 +55,104 @@ shapefiles:
     url: http://s3.amazonaws.com/tilezen-assets/curated/admin_areas_20180409.zip
 
   - name: ne_110m_lakes
-    url: http://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/physical/ne_110m_lakes.zip
+    url: https://naciscdn.org/naturalearth/5.1.0/110m/physical/ne_110m_lakes.zip
```
Comment (nvkelso) [inline data/assets.yaml:58]: Thanks for switching these to versioned CDN links... much more obvious which version we're on and makes updating NE a deliberate process.

File: data/functions.sql [hunk 3]
```diff
@@ -1006,28 +1006,30 @@ BEGIN
...
+        WHERE t.wikidataid = wikidata_id;
+      END IF;
+
+      IF NOT FOUND THEN
+        RETURN '{}'::jsonb;
+      END IF;
   END IF;
 
-  -- try states and provinces
-  IF NOT FOUND THEN
```
Comment (nvkelso) [inline data/functions.sql:1025]: Don't you want to keep this in here to bail early if found?

File: data/functions.sql [hunk 4]
```diff
@@ -1051,6 +1053,65 @@ BEGIN
...
+    END IF;
+
+    IF NOT FOUND THEN
+        RETURN '{}'::jsonb;
+    END IF;
+  END IF;
+
+  -- There is no label_x and label_y for the non-countries
+  SELECT
+    fclass_iso, fclass_tlc, NULL, NULL INTO fclass_iso, fclass_tlc, label_x, label_y
```
Comment (nvkelso) [inline data/functions.sql:1089]: This is confusingly called: "latitude" (label_y) and "longitude" (label_x) in the NE admin-1 table. Please update to pull them out.

File: data/functions.sql [hunk 5]
```diff
@@ -1051,6 +1053,65 @@ BEGIN
...
+  FROM ne_10m_admin_1_states_provinces sp
+  WHERE sp.wikidataid = wikidata_id;
+
+  -- finally, try localities
+  IF NOT FOUND THEN
+  SELECT
+    fclass_iso, fclass_tlc, NULL, NULL INTO fclass_iso, fclass_tlc, label_x, label_y
```
Comment (nvkelso) [inline data/functions.sql:1096]: Same deal: This is confusingly called: "latitude" (label_y) and "longitude" (label_x) in the NE populated places table. Please update to pull them out.

**Review by nvkelso — COMMENTED** [review]
Can you add NE and OSM unit tests for TLC, please?

File: data/functions.sql (hunk 4 above)
Comment (nvkelso) [reply data/functions.sql:1089]: (so keep the output as label_{x,y} but the input would be `longitude, latitude`

File: data/functions.sql (hunk 3 above)
Comment (travisgrigsby) [reply data/functions.sql:1025]: You're right.  I thought I was returning on green line 1024, but that's only if we didn't find it.  Updated to check if it's a country else do something else, then return once at the end.

File: data/assets.yaml (hunk 2 above)
Comment (travisgrigsby) [reply data/assets.yaml:58]: I agree.  Much clearer

File: data/functions.sql (hunk 5 above)
Comment (travisgrigsby) [reply data/functions.sql]: fixed in 31a5b4a27b76b89b24214157b4758b3ddebad030

File: data/functions.sql (hunk 4 above)
Comment (travisgrigsby) [reply data/functions.sql]: fixed in 31a5b4a27b76b89b24214157b4758b3ddebad030

File: data/functions.sql (hunk 3 above)
Comment (travisgrigsby) [reply data/functions.sql:1025]: Fixed in 31a5b4a27b76b89b24214157b4758b3ddebad030

File: data/assets.yaml (hunk 1 above)
Comment (travisgrigsby) [reply data/assets.yaml]: updated to most recent (which I think @jeffdefacto found an issue in for the iso countries.zip)

File: data/assets.yaml [hunk 6]
```diff
@@ -141,12 +141,15 @@ shapefiles:
 
   - name: ne_10m_admin_0_countries
     url: https://naciscdn.org/naturalearth/5.1.0/10m/cultural/ne_10m_admin_0_countries.zip
+    prj: 3857
```
Comment (nvkelso) [inline data/assets.yaml:144]: These files aren't web mercator (3857). Instead they need to be re-projected into web mercator from WGS84 lat/lng

File: data/assets.yaml (hunk 1 above)
Comment (nvkelso) [reply data/assets.yaml]: That zero length char definition is really odd. Opens fine in QGIS but PostGIS barfs on it (saying it doesn't support 0 length varchar fields).

Filed https://github.com/mbloch/mapshaper/issues/541 upstream.

Comment (nvkelso) [reply data/assets.yaml]: With https://github.com/mbloch/mapshaper/issues/541 fixed upstream, I'll update Natural Earth to v5.1.1 with the fix.

File: data/assets.yaml (hunk 2 above)
Comment (nvkelso) [reply data/assets.yaml:58]: These will all need to switch to 5.1.1 to pick up the other change.

File: data/assets.yaml (hunk 1 above)
Comment (nvkelso) [reply data/assets.yaml]: Natural Earth 5.1.1 is now out, please update here and see if the problem persists.

File: data/assets.yaml (hunk 6 above)
Comment (travisgrigsby) [reply data/assets.yaml]: updated here: 2635448740ded3a712fbb9cc5c61d108ac734ec9

**Review by nvkelso — APPROVED** [review]

**Pull request merged**